docker/build.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -e
+# Copy the config files into the Docker folder
+rm -rf config && cp -r ../config ./config
+
+# Get number of cores available on the machine. Build with all cores but one, to prevent RAM choking
+cores_available=$(nproc --all)
+n_build_cores=$(( cores_available -1 ))
+
+# Get CUDA architecture, if GPUs are available
+cat << EOF > ./get_gpu_arch.cu
+#include <stdio.h>
+int main() {
+  int count = 0;
+  if (cudaSuccess != cudaGetDeviceCount(&count)) return -1;
+  if (count == 0) return -1;
+  for (int device = 0; device < count; ++device) {
+    cudaDeviceProp prop;
+    if (cudaSuccess == cudaGetDeviceProperties(&prop, device))
+      printf("%d ", prop.major*10+prop.minor);
+  }
+  return 0;
+}
+EOF
+gpu_arch_codes=""
+if command -v nvcc &> /dev/null
+then
+  nvcc ./get_gpu_arch.cu -o ./get_gpu_arch
+  gpu_arch_codes=$(./get_gpu_arch)
```

Comment:
Should this not be quoted? In general I'd rather have all subshell invocations quoted even if it's unnecessary just to avoid errors, though I'm happy to be convinced otherwise